flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/WatermarkAssignerChangelogNormalizeTransposeRule.java

```diff
@@ -359,6 +360,16 @@ private final RelNode buildTreeInOrder(
                                         inputNode,
                                         nodeAndTrait.f1.getTrait(
                                                 FlinkRelDistributionTraitDef.INSTANCE()));
+            } else if (currentNode instanceof StreamPhysicalChangelogNormalize) {
```

Comment:
This is hard to maintain now, I create an issue to do the refactor: https://issues.apache.org/jira/browse/FLINK-29981